a10_octavia/tests/unit/controller/worker/tasks/test_nat_pool_tasks.py

```diff
@@ -0,0 +1,160 @@
+
+try:
+    from unittest import mock
+except ImportError:
+    import mock
+
+from octavia.common import data_models as o_data_models
+
+from a10_octavia.common.data_models import VThunder
+import a10_octavia.controller.worker.tasks.nat_pool_tasks as task
+from a10_octavia.tests.common import a10constants
+from a10_octavia.tests.unit import base
+
+VTHUNDER = VThunder()
+LB = o_data_models.LoadBalancer(id=a10constants.MOCK_LOAD_BALANCER_ID)
+
+
+class TestHandlerNatPoolTasks(base.BaseTaskTestCase):
+
+    def setUp(self):
+        super(TestHandlerNatPoolTasks, self).setUp()
+        imp.reload(task)
+        self.client_mock = mock.Mock()
+
+    def tearDown(self):
+        super(TestHandlerNatPoolTasks, self).tearDown()
+
+    def test_create_nat_pool_with_nat_pool_not_exists_with_nat_pool_flavor(self):
+        flavor = {u'nat_pool': {u'start_address': u'172.16.1.101', u'end_address': u'172.16.1.108',
+                                u'netmask': u'/24', u'gateway': u'172.16.1.1',
+                                u'pool_name': u'pool1'}}
+
+        vthunder = copy.deepcopy(VTHUNDER)
+        mock_nat_pool_task = task.NatPoolCreate()
+        mock_nat_pool_task.axapi_client = self.client_mock
+        self.client_mock.nat.pool.exists.return_value = False
+        self.client_mock.nat.pool.try_get.return_value = None
+        mock_nat_pool_task.execute(LB, vthunder, flavor_data=flavor)
+        args, kwargs = self.client_mock.nat.pool.create.call_args
+        self.assertEqual(kwargs['pool_name'], 'pool1')
+        self.assertEqual(kwargs['start_address'], '172.16.1.101')
+        self.assertEqual(kwargs['gateway'], '172.16.1.1')
+
+    def test_create_nat_pool_with_nat_pool_exists_with_nat_pool_flavor(self):
+        flavor = {u'nat_pool': {u'start_address': u'172.16.1.101', u'end_address': u'172.16.1.108',
+                                u'netmask': u'/24', u'gateway': u'172.16.1.1',
+                                u'pool_name': u'pool1'}}
+
+        device_pool = {u'pool': {u'uuid': u'947c228e-443a-11eb-9273-525400bf462d',
+                                 u'start-address': u'172.16.1.101', u'port-overload': 0,
+                                 u'a10-url': u'/axapi/v3/ip/nat/pool/pool1', u'netmask': u'/24',
+                                 u'end-address': u'172.16.1.108', u'ip-rr': 0,
+                                 u'gateway': u'172.16.1.1', u'pool-name': u'pool1'}}
+
+        vthunder = copy.deepcopy(VTHUNDER)
+        mock_nat_pool_task = task.NatPoolCreate()
+        mock_nat_pool_task.axapi_client = self.client_mock
+        self.client_mock.nat.pool.exists.return_value = True
+        self.client_mock.nat.pool.try_get.return_value = device_pool
+        mock_nat_pool_task.execute(LB, vthunder, flavor_data=flavor)
+        self.client_mock.nat.pool.create.not_called()
+
+    def test_create_nat_pool_with_nat_pool_not_exists_with_nat_pool_list_flavor(self):
+        flavor = {u'nat_pool_list': [{u'start_address': u'172.16.2.101',
```

Comment:
I would suggest global variable as `FLAVOR1` and `FLAVOR2` and then `FLAVOR_LIST = {u'nat_pool_list' : [ FLAVOR1, FLAVOR2] }`

Reply:
ok